app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -94,9 +96,17 @@ protected void onStart() {
     public void onBackPressed() {
         if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
             binding.drawerLayout.closeDrawer(GravityCompat.START);
-        } else {
-            backPressHandler.onBackPressed(this, super::onBackPressed);
         }
+        else {
+            if (fragmentManager.findFragmentById(R.id.fragment_container) instanceof EventListFragment ||
+                fragmentManager.findFragmentById(R.id.fragment_container) instanceof EventDashboardFragment) {
+                backPressHandler.onBackPressed(this, super::onBackPressed);
```

Comment:
Again, "checking instance" is poor solution. You can see yourself how highly dependent it is on the current structure of app. Code should never even know what the structure of app is. Anytime I change these fragments or add new ones, I'll have to change this code. Which has nothing to do with these fragments